NBitcoin/Protocol/Node.cs

```diff
@@ -1074,41 +1087,57 @@ public void RespondToHandShake(CancellationToken cancellation = default(Cancella
 			}
 		}
 
-		public void Disconnect()
-		{
-			Disconnect(null, null);
-		}
-
 		int _Disconnecting;
 
-		public void Disconnect(string reason, Exception exception = null)
+		/// <summary>
+		/// Disconnects the node and checks the listener thread.
+		/// </summary>
+		public void Disconnect(string reason = null)
 		{
-			DisconnectAsync(reason, exception);
-			AssertNoListeningThread();
-			_Connection.Disconnected.WaitOne();
-		}
+			if (IsConnected == false)
+				return;
 
-		private void AssertNoListeningThread()
-		{
-			if(_Connection._ListenerThreadId == Thread.CurrentThread.ManagedThreadId)
-				throw new InvalidOperationException("Using Disconnect on this thread would result in a deadlock, use DisconnectAsync instead");
+			DisconnectNode(reason, null);
+
+			try
+			{
+				AssertListenerThreadIsNotCurrentThread();
+				_Connection.Disconnected.WaitOne();
+			}
+			catch (InvalidOperationException)

```

Comment:
No need to use catch

Reply:
The catch is still needed as AssertListenerThreadIsNotCurrentThread throws an exception which I need to catch and rethrow but STILL call the finalize method on NodeConnection .

Reply:
I believe you can remove the catch and only keep the finally no?